File: Python/polinomial_regression.py

#Regresion Polinomica

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Cargar Dataset
dataset = pd.read_csv('Position_Salaries.csv')

#Definir Variables X y y
X = dataset.iloc[: , 1:2].values
y = dataset.iloc[: , 2].values

#Regresion Lineal simple (Para Ver Diferencias)
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X,y)

#la Regresion Polinomica t
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly =  poly_reg.fit_transform(X)
lin_reg2 = LinearRegression()
lin_reg2.fit(X_poly, y)


#Visualizacion de Regresion lineal simple
plt.scatter(X, y , color ="red")
plt.plot(X, lin_reg.predict(X), color = "blue")
plt.title("Modelo de Regresion lineal Simple")
plt.xlabel("Posicion del empleado")
plt.ylabel("Sueldo del Empleado")
plt.show()

#Visualizacion de Regresion Polinomica
X_grid =np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid),1)
plt.scatter(X, y , color ="red")
plt.plot(X_grid, lin_reg2.predict(poly_reg.fit_transform(X_grid)), color = "blue")
plt.title("Modelo de Regresion lineal Polinomial")
plt.xlabel("Posicion del empleado")
plt.ylabel("Sueldo del Empleado")
plt.show()

#Prediccion de nuestros modelos

lin_reg.predict([[6.5]])
lin_reg2.predict(poly_reg.fit_transform([[6.5]]))

File: R/polinomial_regression.R
# Regresion polinomica

dataset = read.csv('Position_SAlaries.csv')
dataset = dataset [, 2:3]

#Regresion Lineal Simple
lin_reg = lm(formula = Salary ~.,
             data = dataset)


#Regresion Polinomial
dataset$Level2 = dataset$Level ^2
dataset$Level3 = dataset$Level ^3
dataset$Level4 = dataset$Level ^4
poly_reg = lm(formula =  Salary ~ .,
              data =  dataset)

#Visualizacion de Regresion Lineal
library(ggplot2)
ggplot() + 
  geom_point(aes(x= dataset$Level, y = dataset$Salary),
             color = "red")+
  geom_line(aes(x =dataset$Level, y=predict(lin_reg, newdata = dataset)),
           color = "blue")+
  ggtitle("Prediccion de Regresion Lineal")+
  xlab("Nivel de Empleado")+
  ylab("Salario")

#Visualizacion de Regresion Polinomica
ggplot() + 
  geom_point(aes(x= dataset$Level, y = dataset$Salary),
             color = "red")+
  geom_line(aes(x =dataset$Level, y=predict(poly_reg, newdata = dataset)),
            color = "blue")+
  ggtitle("Prediccion de Regresion Polinomica")+
  xlab("Nivel de Empleado")+
  ylab("Salario")


#Prediccion de Nuevos Datos con Regresion Lineal
y_pred = predict(lin_reg, newdata = data.frame(Level = 6.5))





#Prediccion con Nuevos Datos con Regresion Polinomica
y_pred_poly = predict(poly_reg, newdata = data.frame(Level = 6.5,
                                                     Level2 = 6.5^2,
                                                     Level3 = 6.5^3,
                                                     Level4 = 6.5^4))


